exporter/loadbalancingexporter/resolver_static.go

```diff
@@ -49,13 +50,23 @@ func (r *staticResolver) start(ctx context.Context) error {
 	return err
 }
 
-func (r *staticResolver) shutdown(_ context.Context) error {
+func (r *staticResolver) shutdown(ctx context.Context) error {
+	r.endpoints = nil
+
+	r.mx.Lock()
+	defer r.mx.Unlock()
+	for _, callback := range r.onChangeCallbacks {
+		callback(r.endpoints)
+	}
```

Comment:
This solution could use some discussion or feedback. The resolver was [specifically designed to only be called once](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/1412#discussion_r518319655), as it's for a static list backends that shouldn't change.

However, the resolver [starts each passed in exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e2d601acb20fe0fa1c99d860ff484a208341af27/exporter/loadbalancingexporter/loadbalancer.go#L136), and due to the `once.Do` functionality, will never shutdown these started exporters. To use existing functionality as much as possible, if we clear the static resolver's list of endpoints and call the `onChangeCallbacks` methods, the existing exporters will be shutdown.

I added the mutex to make sure we're not calling `resolve` concurrently, but there may be a cleaner way to do this. The requirement here is that `shutdown` is called after the initial `start` method calls `resolve`.

Reply:
I can file an issue for this if it's preferable to discuss elsewhere, just let me know.

Reply:
I think the approach here is fine, thank you for catching this!

Reply:
Why do we need an extra lock here? I see `resolve` is called synchronously at the start of the load balancer. I don't think we allow calling Start and Shutdown concurrently.

Reply:
I believe this was simply done to be safe. If it's more confusing than helpful I can remove it.

Reply:
I would prefer not introducing locks that are not needed

Reply:
I've removed the mutex.